src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1400,6 +1400,7 @@ private void EnterParameters(ImmutableArray<ParameterSymbol> parameters)
 
         protected virtual void EnterParameter(ParameterSymbol parameter)
         {
+            Visit(parameter.ExplicitDefaultExpression);
```

Comment:
>Visit(parameter.ExplicitDefaultExpression); [](start = 12, length = 43)

This should be visited as an RValue (VisitRvalue). Otherwise, if the default value is a logical expression, we might end up with a split state, which is likely to cause a crash later. Please add a unit-test for the broken scenario before applying the fix.